src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  createConcurrentSafePromise,
+  addQueryID,
+  debounce,
+  addAbsolutePosition,
+  noop,
+  escapeHits,
+} from '../../lib/utils';
+import {
+  Connector,
+  Hits,
+  Hit,
+  FindAnswersOptions,
+  FindAnswersResponse,
+} from '../../types';
+
+type IndexWithAnswers = {
+  readonly findAnswers: any;
+};
+
+function hasFindAnswersMethod(
+  answersIndex: IndexWithAnswers | any
```

Comment:
```suggestion
  answersIndex: object
```

since it's a type guard, it should accept as broad as possible, in this case an object (not any) but asserts it has answers

Reply:
maybe it could even accept `ReturnType<SearchClient['initIndex']>` as argument, and only assert that findAnswers exists

Reply:
```
type IndexWithAnswers = {
  readonly findAnswers: any;
} & ReturnType<Exclude<SearchClient['initIndex'], undefined>>;

function hasFindAnswersMethod(
  answersIndex: ReturnType<Exclude<SearchClient['initIndex'], undefined>>
): answersIndex is IndexWithAnswers {
  return typeof (answersIndex as IndexWithAnswers).findAnswers === 'function';
} 
```

I tried this but got this

![image](https://user-images.githubusercontent.com/499898/106778062-dd821b00-6645-11eb-9508-21a01150b290.png)

Anything related to "answers" will fail in client v3 because it doesn't have `findAnswers` method and related types. The current version is what @yannickcr and I got so far, but let's see next time if it can be improved.